glean-core/ffi/src/lib.rs

```diff
@@ -180,6 +186,36 @@ pub extern "C" fn glean_enable_logging() {
     }
 }
 
+#[cfg(all(not(target_os = "android"), not(target_os = "ios")))]
+static FD_LOGGER: OnceCell<fd_logger::FdLogger> = OnceCell::new();
+
+/// Initialize the logging system to send JSON messages to a pipe.
+///
+/// `fd` is a writable file descriptor (on Unix) or file handle (on Windows).
+///
+/// # Safety
+/// Unsafe because the fd u64 passed in will be interpreted as either a file
+/// descriptor (Unix) or file handle (Windows) without any checking.
+#[cfg(all(not(target_os = "android"), not(target_os = "ios")))]
+#[no_mangle]
+pub unsafe extern "C" fn glean_enable_logging_to_pipe(fd: u64) {
```

Comment:
Given that this function _does_ land in the generated header file, should we have an empty implementation for Android/iOS?
On the other hand if we ever end up using it (by accident) on these platforms it will fail at load time with an unresolved symbol.

At the very least we should document it that it's not available everywhere.

Reply:
+1 on documenting.

I think there's some feature in cbindgen to wrap things in C `#define`s -- cbindgen warns about that, but I couldn't find the docs on that feature anywhere.  Ultimately, I gave up (but should have mentioned that here).